docs/abstract.rst

```diff
@@ -0,0 +1,258 @@
+.. _abstract_widget_intro:
+
+Understanding BaseImageWidget
+=============================
+
+``astrowidgets`` provides an abstract class called
+`~astrowidgets.core.BaseImageWidget` to allow developers from different
+visualization tools (hereafter known as "backends") to implement their own
+solutions using the same set of API. This design is based on
+`nb-astroimage-api <https://github.com/eteq/nb-astroimage-api>`_, with the
+goal of making all functionality available by a compact and clear API.
+This API-first approach would allow manipulating the view programmatically
+in a reproducible way.
+
+The idea of the abstract class is that ``astrowidgets`` users would be able
+to switch to the backend of their choice without much refactoring of their own.
+This would enable, say, astronomers with different backend preferences to
+collaborate more easily via Jupyter Lab/Notebook.
+
+The following sub-sections lay out the envisioned high-level usage of
+``astrowidgets`` regardless of backend inside Jupyter Lab/Notebook.
+However, the examples are not exhaustive. For the full API definition,
+please see :ref:`abstract_api`.
+
+.. _abstract_viewer:
+
+Creating a Viewer
+-----------------
+
+The snippet below is all you should need to make an image widget.
+The widget should be a part of the
+`ipywidgets framework <https://ipywidgets.rtfd.io>`_ so that it can
+be easily integrated with other controls:
+
+.. code-block:: python
+
+    from astrowidgets.somebackend import ImageWidget
```

Comment:
Conceptually I wonder if this is better presented as
```suggestion
    from somevizpackage.astrowidget_backend import ImageWidget
```
? That is, make it clear that external packages can and should write their own backends, rather than it being an import from astrowidgets?